nixos/modules/services/networking/go-camo.nix

```diff
@@ -0,0 +1,52 @@
+{ lib, pkgs, config, ... }:
+with lib;
+
+let
+  cfg = config.services.go-camo;
+in
+{
+  options.services.go-camo = {
+    enable = mkEnableOption "go-camo service";
+    addressPort = mkOption {
+      type = types.nullOr types.str;
+      default = null;
+      description = "Address:Port to bind to for HTTP (default: 0.0.0.0:8080)";
+      apply = v: optionalString (v != null) "--listen=${v}";
+    };
+    sslListen = mkOption {
+      type = types.nullOr types.str;
+      default = null;
+      description = "Address:Port to bind to for HTTPS/SSL/TLS";
+      apply = v: optionalString (v != null) "--ssl-listen=${v}";
+    };
+    sslKey = mkOption {
+      type = types.nullOr types.path;
+      default = null;
+      description = "SSL private key path";
+      apply = v: optionalString (v != null) "--ssl-key=${v}";
+    };
+    sslCert = mkOption {
+      type = types.nullOr types.path;
+      default = null;
+      description = "SSL certificate path";
+      apply = v: optionalString (v != null) "--ssl-cert=${v}";
+    };
+    key = mkOption {
+      type = types.str;
+      apply = v: "--key=${v}";
+      description = "HMAC key";
+    };
+    extraOptions = mkOption {
+      type = types.str;
+      default = "";
+      description = "Remaining options passed to the go-camo command";
+    };
+  };
+
+  config = mkIf cfg.enable {
+    systemd.services.go-camo = {
+      wantedBy = [ "multi-user.target" ];
+      serviceConfig.ExecStart = "${pkgs.go-camo}/bin/go-camo ${cfg.addressPort} ${cfg.key} ${cfg.sslListen} ${cfg.sslKey} ${cfg.sslCert} ${cfg.extraOptions}";
```

Comment:
Does that work if the options are empty?

Reply:
Extra options? Yes.
All options? The key is required so there's always at least one.